### PR TITLE
Refactor `KeycloakDeploymentConfig`

### LINKED ISSUE #14879: Refactor `KeycloakDeploymentConfig`
Make mapping Quarkus dist options more generic

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #14879

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java [hunk 1]
```diff
@@ -280,10 +255,65 @@ protected void assumeFirstClassCitizens(KeycloakStatusBuilder status) {
...
+
+            return this;
+        }
+
+        public <R> OptionMapper<T> mapOption(String optionName, R optionValue) {
+            return mapOption(optionName, s -> optionValue);
+        }
+
+        protected <R extends Collection<?>> OptionMapper<T> mapOptionFromCollection(String optionName, Function<T, R> optionValueSupplier) {
+            return mapOption(optionName, s -> optionValueSupplier.apply(s).stream().map(String::valueOf).collect(Collectors.joining(",")));
```
Comment (mabartos) [inline operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java:316]:
```suggestion
            return mapOption(optionName, s -> {
                var value = optionValueSupplier.apply(s);
                if (value == null) return null;
                return value.stream().map(String::valueOf).collect(Collectors.joining(","));
            });
```
The NPE can be thrown here. For instance, when only the enabled features are specified, but not disabled ones. Maybe not so nice code, but I think it's pretty good readable, and intuitive. WDYT? 

Probably would be good to add some test cases for that kind of logic.

**Review by mabartos — CHANGES_REQUESTED** [review]
@vmuzikar Thanks for the PR! Nice work! Almost everything works as expected, but only one thing.

File: operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java (hunk 1 above)
Comment (vmuzikar) [reply operator/src/main/java/org/keycloak/operator/controllers/KeycloakDistConfigurator.java]: Good catch, thanks! Commited your suggestion and added a test for it.

**Review by mabartos — APPROVED** [review]
@vmuzikar LGTM! I believe once the GHAs pass, this can be merged.

**Pull request merged**